tutorials/nextjs-app-serverless-functions-sqldb/index.mdx

```diff
@@ -0,0 +1,948 @@
+          </Container>
+        </main>
+      );
+    }
+    
+    type Params = {
+      params: {
+        slug: string;
+      };
+    };
+    
+    export async function generateMetadata({ params }: Params): Promise<Metadata> {
+      const post = await getPostBySlug(params.slug);
+    
+      if (!post) {
+        return notFound();
+      }
+    
+      const title = `${post.title} | Next.js Blog Example with ${CMS_NAME}`;
+    
+      return {
+        openGraph: {
+          title,
+          images: [post.ogImage.url],
+        },
+      };
+    }
+    
+    export async function generateStaticParams() {
+      const posts = await getAllPosts();
+    
+      return posts.map((post) => ({
+        slug: post.slug,
+      }));
+    }
+    ```
+  
+4. Run the following command to ensure that you can run the application locally:
+    ```bash
+    npm run dev
+    ```
+   
+5. Go to [localhost:3000](http://localhost:3000) in your browser. The images and titles of the first blog post have been updated with the content from your database.
+   
+Congratulations! You can already deploy your application by building a new container version, pushing it to the **Container Registry** and deploying it to **Serverless Containers**.
+
+However, your application would then connect with your [user credentials](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#user), which is not a recommended security practice.
+To secure your deployment, we will now add a dedicated [IAM application](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#application), give it the minimum required permissions, and provide its credentials to your application.
+
+## Secure and redeploy your application
+
+1. Run the following command to create an [IAM application](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#application) and export it as a variable: 
+    ```bash
+    export SCW_APPLICATION_ID=$(scw iam application create name=tutorial-nextjs-app -o json | jq -r '.id')
+    ```
+    The `SCW_APPLICATION_ID` environment variable will store the IAM application ID so you can use it in later commands.
+
+2. Create an [IAM policy](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#policy) giving your application rights to access the database:
+    ```bash
+    scw iam policy create application-id=$SCW_APPLICATION_ID name=tutorial-nextjs-policy rules.0.organization-id=$(scw config get default-organization-id) rules.0.permission-set-names.0=ServerlessSQLDatabaseFullAccess
+    ```
+    The permission `ServerlessSQLDatabaseFullAccess` lets your application read and write data or create new databases. You can restrict this later to fit your database use cases, using `ServerlessSQLDatabaseRead` or `ServerlessSQLDatabaseReadWrite` permissions sets.
+
+3. Create an [IAM Secret Key](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#api-key):
```

Comment:
```suggestion
3. Create an [IAM Secret Key](/identity-and-access-management/iam/concepts/#api-key):
```

Reply:
Should be an API key here, you can't create a secret key on its own without creating an API key. You could specify that they will only need the API key's secret key though!